File: RoRClient/Commands/Editor/Move/MoveCraneCommand.cs
using RoRClient.Commands.Base;
using RoRClient.Communication.DataTransferObject;
using RoRClient.Models.Game;
using RoRClient.Models.Session;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RoRClient.Commands.Editor.Move
{
    class MoveCraneCommand : CommandBase
    {
        private int oldXPos;
        private int oldYPos;
        private int newXPos;
        private int newYPos;


        public MoveCraneCommand(RoRSession session, MessageInformation message) : base(session, message)
        {
            this.newXPos = message.GetValueAsInt("newXPos");
            this.newYPos = message.GetValueAsInt("newYPos");
        }

        public override void Execute()
        {
            EditorSession editor = (EditorSession) session;
            Square square = editor.Map.GetSquare(newXPos, newYPos);
            Rail rail = (Rail)square.PlaceableOnSquare;
           
            Crane crane = (Crane)rail.PlaceableOnRail;
            crane.Square = editor.Map.GetSquare(newXPos, newYPos);
            rail.PlaceableOnRail = null;
            rail.PlaceableOnRail = crane;
            Console.WriteLine("newX" + crane.Square.PosX + "newY:"+crane.Square.PosY );
        }
    }
}
 

File: RoRServer/src/main/java/commands/game/MoveCraneCommand.java
package commands.game;

import java.util.UUID;

import commands.base.Command;
import communication.MessageInformation;
import models.game.Compass;
import models.game.Crane;
import models.game.Stock;
import models.game.Trainstation;
import models.session.GameSession;
import models.session.RoRSession;

public class MoveCraneCommand implements Command{
	
	private int xPos;
	private int yPos;
	private UUID stockId;
	private Stock stock;
	private Trainstation trainstation;
	private Crane crane;
	protected GameSession session;
	
	public MoveCraneCommand(RoRSession session, MessageInformation messageInfo) { 

		this.xPos =  messageInfo.getValueAsInt("posX");
		this.yPos =  messageInfo.getValueAsInt("posY");
		this.session = (GameSession)session;
		this.stock = (Stock)session.getMap().getPlaceableOnSquareById(messageInfo.getValueAsUUID("stockId"));
		this.trainstation = (Trainstation)session.getMap().getPlaceableOnSquareById(this.stock.getTrainstationId());
		
		this.crane = this.trainstation.getCrane();
		
	}

	@Override
	public void execute() {
		
		this.crane.moveToTakeTheGoods();
		System.out.println("I'm moving for gooood(Crane)");
		
	}

}
